From 2453c89ead650ab8e21f13c1b0a1b4054e0e7067 Mon Sep 17 00:00:00 2001
From: "Jason M. Bills" <jason.m.bills@linux.intel.com>
Date: Mon, 27 Jul 2020 11:27:47 -0700
Subject: [PATCH] Recalculate AW FCS on WrEndPointConfig command

The WrEndPointConfig command requires recalculating the AW FCS
on retry, so set the has_aw_fcs parameter to true.

Tested:
Ran a WrEndPointConfig command with C-states enabled to confirm
that it correctly retries the command:
peci_cmds WrEndpointConfigPCILocal 0 13 12 0 0x2d4 0x601e0

Signed-off-by: Jason M. Bills <jason.m.bills@linux.intel.com>
---
 drivers/peci/peci-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/peci/peci-core.c b/drivers/peci/peci-core.c
index 426ee2be3271..ba561bf810f9 100644
--- a/drivers/peci/peci-core.c
+++ b/drivers/peci/peci-core.c
@@ -1059,7 +1059,7 @@ static int peci_cmd_wr_end_pt_cfg(struct peci_adapter *adapter, void *vmsg)
 		return -EINVAL;
 	}
 
-	ret = peci_xfer_with_retries(adapter, msg, false);
+	ret = peci_xfer_with_retries(adapter, msg, true);
 
 out:
 	umsg->cc = msg->rx_buf[0];
-- 
2.17.1

